src/Language/Fixpoint/Solver/Sanitize.hs

```diff
@@ -79,6 +80,45 @@ addLiterals si = si { F.dLits = F.unionSEnv (F.dLits si) lits'
   where
     lits'      = M.fromList [ (F.symbol x, F.strSort) | x <- symConsts si ]
 
+
+--------------------------------------------------------------------------------
+-- | `eliminateEta` converts equations of the form f x = g x into f = g
```

Comment:
@yiyunliu so, if it is `f x y = g y` your edit will not apply, correct? Should we change that?

Reply:
Yes, I'm aware of this issue. I made the change conservatively since I wasn't sure if this was the right approach.
I guess we can take the length of the shorter list and compare the truncated lists. I'll fix that tomorrow.